chia/wallet/wallet.py

```diff
@@ -319,19 +320,23 @@ async def _generate_unsigned_transaction(
         max_coin_amount: Optional[uint64] = None,
         exclude_coin_amounts: Optional[List[uint64]] = None,
         exclude_coins: Optional[Set[Coin]] = None,
+        puzzle_decorator_override: Optional[List[Dict[str, Any]]] = None,
         reuse_puzhash: Optional[bool] = None,
     ) -> List[CoinSpend]:
         """
         Generates a unsigned transaction in form of List(Puzzle, Solutions)
         Note: this must be called under a wallet state manager lock
         """
+        decorator_manager: PuzzleDecoratorManager = self.wallet_state_manager.decorator_manager
+        if puzzle_decorator_override is not None:
+            decorator_manager = PuzzleDecoratorManager.create(puzzle_decorator_override)
+
         primaries = []
-        if (primaries_input is None and amount > 0) or primaries_input is not None:
-            primaries.append(Payment(newpuzzlehash, amount, [] if memos is None else memos))
         if primaries_input is not None:
```

Comment:
```suggestion
        if (primaries_input is None and amount > 0) or primaries_input is not None:
            primaries.append(Payment(newpuzzlehash, amount, [] if memos is None else memos))
        if primaries_input is not None:
```

This is just adding back what you removed to show what i mean as part of the other comment.